xapian-letor/api/ranker.cc

```diff
@@ -37,11 +37,15 @@
 using namespace std;
 using namespace Xapian;
 
-struct scoreComparer {
-    bool operator()(const pair<Xapian::docid, int>& first_pair, const pair<Xapian::docid, int>& second_pair) const {
-        return first_pair.second > second_pair.second;
-    }
-};
+bool
+Ranker::ScoreComparer::operator()(const FeatureVector & firstfv, const FeatureVector& secondfv) const {
+    return firstfv.get_score() > secondfv.get_score();
+}
```

Comment:
That's still not a reason to make them into functors. They could just be non-static functions, couldn't they? The only reason to make them into functors is to allow other versions to be swapped in, and I'm not sure why they might be.